tools/check_release.py

```diff
@@ -0,0 +1,447 @@
+# > python tools/check_release.py 
+#
+# It will look for local clones of the test app repos. If they don't exist
+# it will clone them. It will then read the latest versions of mbed and mbed-dev
+# (an assumption is made that both of these are already cloned in your Mercurial area).
+# The lib files within the test apps are then updated to the corresponding version in 
+# the associated lib itself. The test apps are then committed and pushed back to the users
+# fork.
+# The test apps will then be compiled for all supported targets and a % result output at the 
+# end.
+#
+# Uses the online compiler API at https://mbed.org/handbook/Compile-API
+# Based on the example from https://mbed.org/teams/mbed/code/mbed-API-helper/
+ 
+ 
+import os, getpass, sys, json, time, requests, logging
+from os.path import dirname, abspath, basename, join
+import argparse
+import subprocess
+import re
+import hglib
+import argparse
+
+# Be sure that the tools directory is in the search path
+ROOT = abspath(join(dirname(__file__), ".."))
+sys.path.insert(0, ROOT)
+
+from tools.build_api import get_mbed_official_release
+
+OFFICIAL_MBED_LIBRARY_BUILD = get_mbed_official_release('2')
+
+def get_compilation_failure(messages):
+    """ Reads the json formatted 'messages' and checks for compilation errors.
+        If there is a genuine compilation error then there should be a new message containing
+        a severity field = Error and an accompanying message with the compile error text.
+        Any other combination is considered an internal compile engine failure 
+    Args:
+    messages - json formatted text returned by the online compiler API.
+    
+    Returns:
+    Either a string containing a compilation error or "Internal" to indicate an error with
+    the online IDE API itself.
+
+    """
+    for m in messages:
+        if 'severity' in m and 'message' in m:
+            if m['severity'] == 'error':
+                return m['message']
+
+    return "Internal"
+                 
+def invoke_api(payload, url, auth, polls, begin="start/"):
+    """ Sends an API command request to the online IDE. Waits for a task completed 
+        response before returning the results.
+
+    Args:
+    payload - Configuration parameters to be passed to the API
+    url - THe URL for the online compiler API
+    auth - Tuple containing authentication credentials
+    polls - Number of times to poll for results
+    begin - Default value = "start/", start command to be appended to URL
+    
+    Returns:
+    result - True/False indicating the success/failure of the compilation
+    fail_type - the failure text if the compilation failed, else None
+    """
+
+    # send task to api
+    logging.debug(url + begin + "| data: " + str(payload))
+    r = requests.post(url + begin, data=payload, auth=auth)
+    logging.debug(r.request.body)
+    
+    if r.status_code != 200:
+        raise Exception("Error while talking to the mbed API")
```

Comment:
That's from Sarah's original script and as I haven't actually seen that hit, I'm not sure what the actual exception is. Http code 200 means " the action requested by the client was received, understood, accepted, and processed successfully" , so not 200 means the request failed to be one of those options , ie a general communication with the API error.